src/Api/Service/RevisionSaver.php

```diff
@@ -74,9 +74,13 @@ public function save( Revision $revision ) {
 		$entityId = $entity->getId();
 		if( !is_null( $entityId ) ) {
 			$params['id'] = $entityId->getSerialization();
-			if( $entity->isEmpty() ) {
-				$params['clear'] = 'true';
-			}
+
+			// Always clear so that removing elements is possible
+			$params['clear'] = 'true';
+			// Add more detail to the default "Cleared an entity" summary
+			// Note: this is later overridden if a summary is provided in the EditInfo
+			$params['summary'] = 'Edited an ' . $entity->getType();
```

Comment:
This will be appended to the "Cleared an entity" summary, right? Looks a bit awkward then and should perhaps be fixed in Wikibase.git